card-tricks/ena/card_tricks.go

```diff
@@ -0,0 +1,43 @@
+package cards
+
+func indexInBound(slice []int, index int) bool {
+	return index >= 0 && index < len(slice)
+}
+
+func GetItem(slice []int, index int) (int, bool) {
+	if indexInBound(slice, index) {
+		return slice[index], true
+	}
+	return 0, false
+}
+
+func SetItem(slice []int, index, value int) []int {
+	if indexInBound(slice, index) {
+		slice[index] = value
+	} else {
+		slice = append(slice, value)
+	}
+	return slice
+}
+
+func PrefilledSlice(value, length int) []int {
+	if length < 1 {
+		return []int{}
+	}
+	ret := make([]int, length)
+	for i, _ := range ret {
+		ret[i] = value
+	}
+	return ret
```

Comment:
Using named returns (I know I know :sweat_smile:) and a constrained loop you can change all of this by:
```
for i := 0; i < length; i++ {
	slice = append(slice, value)
}

return
```
I think it's cleaner and more easy to maintain.

Reply:
Now it’s better, but a named return would save you one line. In functions as short as this one, I think it’s ok.

Reply:
Yeah, named parameters is a nice feature, in particular if returning more than one value

Reply:
@davidmogar note that the named return vars are defined to the zero value automatically.
If you were modifying one of your arguments you’d have to copy it to the named return one.

Reply:
@fnasser thats a case by case thing. Let’s say you are appending an element to an slice you are passing as an argument and you want to return it. This is a ver silly example, but you wouldn’t need to copy it (just the copy created by the append call).

But yeah, I get what you mean. Not good to over-do with named returns 😉